modules/omx/module.c

```diff
@@ -0,0 +1,149 @@
+#include "omx.h"
+
+#include <stdlib.h>
+
+#include <re/re.h>
+#include <rem/rem.h>
+#include <baresip.h>
+
+/////// baresip Compatibility Layer
+
+int omx_vidisp_alloc(struct vidisp_st **vp, const struct vidisp* vd, struct vidisp_prm *prm,
+	const char *dev, vidisp_resize_h *resizeh, void *arg);
+int omx_vidisp_display(struct vidisp_st *st, const char *title,
+		   const struct vidframe *frame);
+int omx_vidisp_update(struct vidisp_st *st, bool fullscreen, int orient, const struct vidrect* window);
+void omv_vidisp_hide(struct vidisp_st *st);
+
+
+
+struct vidisp_st {
+	const struct vidisp *vd;  /* inheritance */	
+	struct vidsz size;             /**< Current size          */
+	struct omx_state* omx;
+};
+
+static struct vidisp* vid; // ???
+
+static struct omx_state omx;
```

Comment:
I found that all the display modules seem to have this global pointer to the vidisp struct? So i assumed that all these modules have some kind of global state.
In this case, this thing exists because OMX needs to be initialized from the main thread context, from module_init().
This might be possible to code this in a nicer way, but that's beyond what i can do tonight in a quick code cleanup, without the test platform. ;)